omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/AdministrationLinksResource1_8.java

```diff
@@ -0,0 +1,180 @@
+		if (administrationLinks == null) {
+			throw new ObjectNotFoundException();
+		}
+
+		return administrationLinks;
+	}
+
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		if (rep instanceof DefaultRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("display");
+			description.addProperty(MODULE_TITLE);
+			description.addProperty(LINKS);
+			description.addSelfLink();
+			return description;
+		} else if (rep instanceof FullRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("display");
+			description.addProperty(MODULE_TITLE);
+			description.addProperty(LINKS);
+			description.addSelfLink();
+			return description;
+		} else if (rep instanceof RefRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("display");
+			description.addProperty(MODULE_TITLE);
+			description.addProperty(LINKS);
+			description.addSelfLink();
+			return description;
+		}
+		return null;
+	}
+
+	@PropertyGetter("uuid")
+	public static String getUuid(ModuleAdministrationLinks1_8 instance) {
+		return instance.getModuleId();
+	}
+
+	@PropertyGetter("display")
+	public static String getDisplay(ModuleAdministrationLinks1_8 instance) {
+		return instance.getModuleTitle();
+	}
+
+	@Override
+	public Model getGETModel(Representation rep) {
+		ModelImpl model = (ModelImpl) super.getGETModel(rep);
+		if (rep instanceof DefaultRepresentation || rep instanceof FullRepresentation) {
+			model
+					.property(MODULE_TITLE, new StringProperty())
+					.property(LINKS, new ArrayProperty(new ObjectProperty()));
+		}
+
+		return model;
+	}
+
+	@Override
+	public NeedsPaging<ModuleAdministrationLinks1_8> doGetAll(RequestContext context) throws ResponseException {
+		return new NeedsPaging<ModuleAdministrationLinks1_8>(getAllAdministrationLinks(), context);
+	}
+
+	private ModuleAdministrationLinks1_8 getAdministrationLinksForModule(String moduleId) {
+		List<Extension> adminListsExtensions = moduleFactoryWrapper.getExtensions(ADMIN_LIST_POINT_ID);
+		for (Extension adminListExtension : adminListsExtensions) {
+			if (adminListExtension instanceof AdministrationSectionExt && adminListExtension.getModuleId()
+					.equals(moduleId)) {
+				return getModuleWithAdminLinks(adminListExtension);
+			}
+		}
+
+		return null;
+	}
+
+	private List<ModuleAdministrationLinks1_8> getAllAdministrationLinks() {
+		List<ModuleAdministrationLinks1_8> modulesWithLinksList = new ArrayList<ModuleAdministrationLinks1_8>();
+
+		List<Extension> adminListsExtensions = moduleFactoryWrapper.getExtensions(ADMIN_LIST_POINT_ID);
```

Comment:
Why are you calling moduleFactoryWrapper.getExtensions here and in the method above?

Reply:
Method `getAdministrationLinksForModule` is used when getting links for one module.
Method `getAllAdministrationLinks` is used when getting links for all modules.

Both call `moduleFactoryWrapper.getExtensions()` as there is no provided method to get extensions with a given point id for one module.